lib/rails_cursor_pagination/paginator.rb

```diff
@@ -109,21 +112,46 @@ def ensure_valid_params!(relation, first, after, last, before, order)
         raise ParameterError,
               "`order` must be either :asc or :desc, but was `#{order}`"
       end
+      if first.present? && first.negative?
+        raise ParameterError, "`first` cannot be negative, but was `#{first}`"
+      end
+      if last.present? && last.negative?
+        raise ParameterError, "`last` cannot be negative, but was `#{last}`"
+      end
+      if limit.present? && limit.negative?
+        raise ParameterError, "`limit` cannot be negative, but was `#{limit}`"
+      end
+
+      true
+    end
+
+    # @param limit [Integer, nil]
+    #   Optional, cannot be combined with `last` or `first`
+    # @param first [Integer, nil]
+    #   Optional, cannot be combined with `last` or `limit`
+    # @param after [String, nil]
+    #   Optional, cannot be combined with `before`
+    # @param last [Integer, nil]
+    #   Optional, requires `before`, cannot be combined
+    #   with `first` or `limit`
+    # @param before [String, nil]
+    #   Optional, cannot be combined with `after`
+    def ensure_valid_params_combinations!(first, last, limit, before, after)
```

Comment:
And for completeness sake, please also add the `@raise` YARDoc annotation here:
```suggestion
    #   Optional, cannot be combined with `after`
    #
    # @raise [RailsCursorPagination::ParameterError]
    #   If parameters are combined in an invalid way
    def ensure_valid_params_combinations!(first, last, limit, before, after)
```